### ISSUE TITLE
Send push notifications about the rollover weekend and force-closure of DLC channel

### ISSUE THREAD

**Issue by klochowicz** [get10101/10101#1300]
@holzeis  proposed the following:

### If the position is about to expire and we are in the rollover weekend
(see also https://github.com/get10101/10101/pull/1287/files#diff-f36ed55db10ebb7b64adaf1ba2895679c0254ebd0ce016492f283d4001ac4f84R373-R383) 

- Friday 3pm UTC: Send a notification that the rollover weekend has started and the user should come online to rollover to the next weekend (Sunday 3pm UTC)
- Saturday 3pm UTC: Reminder that the position is about to expire if the user is not coming online to rollover.
- Sunday 12am UTC: Reminder that the position will expire in 3 hours unless the user is coming online to rollover.

### If the position has expired and was not online to close the position.

- At trying to notify about the `AsyncOrder` and the user is not online: Send notification that the position has been expired and the user should come online to close the position.
- For the next week everyday at 3pm UTC: Send a notification that the position has expired and the user should come online or we have to force close the dlc_channel.

### If the position is still expired after a week

Send last notification to the user that we will force close the dlc_channel in due time

_Originally posted by @holzeis in https://github.com/get10101/10101/pull/1297#pullrequestreview-1626483354_

**Comment by klochowicz** [comment]
@holzeis how certain are we that we will *never* change the position expiry? Have we settled for the weekly term and the "rollover weekend" so profoundly that there is no desire to ever backstep on it? 

> At trying to notify about the AsyncOrder and the user is not online: Send notification that the position has been expired and the user should come online to close the position.

This one is not needed, as my current PR detects when a position has expired and tells the user that.

> For the next week everyday at 3pm UTC: Send a notification that the position has expired and the user should come online or we have to force close the dlc_channel.

I would probably not stick to 3pm UTC but just add days to their creation days, otherwise you're just annoying half of the world by waking them up with an irrelevant notification.
other than that, it makes sense.

**Comment by holzeis** [comment]
> @holzeis how certain are we that we will *never* change the position expiry? Have we settled for the weekly term and the "rollover weekend" so profoundly that there is no desire to ever backstep on it? 

Nothing is carved in stone but if we change it we will also have to change the logic.

> > At trying to notify about the AsyncOrder and the user is not online: Send notification that the position has been expired and the user should come online to close the position.
> 
> 
> 
> This one is not needed, as my current PR detects when a position has expired and tells the user that.

No, what your change does is not needed as this is already in main. Your change adds yet another task that constantly checks for expired positions.

I am also nit a fan that the a notification service is responsible for that business logic.

> > For the next week everyday at 3pm UTC: Send a notification that the position has expired and the user should come online or we have to force close the dlc_channel.
> 
> 
> 
> I would probably not stick to 3pm UTC but just add days to their creation days, otherwise you're just annoying half of the world by waking them up with an irrelevant notification.
> 
> other than that, it makes sense.

I am not invested in 3pm, but @bonomat had reasons for it.

**Comment by bonomat** [comment]
> I am not invested in 3pm, but @bonomat had reasons for it.

3pm GMT is when the BRR index is being announced which is being used internationally for settling options and futures: https://www.cfbenchmarks.com/data/indices/BRR

**Comment by bonomat** [comment]
> I would probably not stick to 3pm UTC but just add days to their creation days, otherwise you're just annoying half of the world by waking them up with an irrelevant notification.

These are two things and we could potentially treat them separately: the position can expire at 3pm UTC but we can send notifications to the user depending on their timezone (assuming we know this).

**Comment by klochowicz** [comment]
> > I would probably not stick to 3pm UTC but just add days to their creation days, otherwise you're just annoying half of the world by waking them up with an irrelevant notification.
> 
> These are two things and we could potentially treat them separately: the position can expire at 3pm UTC but we can send notifications to the user depending on their timezone (assuming we know this).



> > I am not invested in 3pm, but @bonomat had reasons for it.
> 
> 3pm GMT is when the CME CF Bitcoin Reference Rate index is being announced which is being used internationally for settling options and futures: https://www.cfbenchmarks.com/data/indices/BRR

thanks for clarification! I thought about it and it makes sense to choose that time, as this is when most of the world are in the same "day". any later time would be too close to "Monday" for e.g. people living in Sydney.

**Comment by klochowicz** [comment]
> > I would probably not stick to 3pm UTC but just add days to their creation days, otherwise you're just annoying half of the world by waking them up with an irrelevant notification.
> 
> These are two things and we could potentially treat them separately: the position can expire at 3pm UTC but we can send notifications to the user depending on their timezone (assuming we know this).

I don't think we know it currently, but we could send that info back during registration etc. however, the easiest heuristics that I tried to explain earlier was "if the user opened a position at a certain time, they're likely be up at the same time before it expires", so that was the idea with offsetting the time from the creation timestamp.

**Comment by holzeis** [comment]
related to https://github.com/get10101/10101/pull/1419

**Comment by bonomat** [comment]
> related to #1419

Thanks for linking this ticket. I think we can close it?

**Comment by holzeis** [comment]
I think we can close it once https://github.com/get10101/10101/pull/1430 lands.
